src/types.rs

```diff
@@ -318,34 +326,42 @@ pub trait Encoding {
     /// or a failure to return the error.
     #[stable]
     fn decode(&self, input: &[u8], trap: DecoderTrap) -> Result<String, Cow<'static, str>> {
+        let mut ret = String::new();
+        self.decode_to(input, trap, &mut ret).map(|_| ret)
+    }
+
+    /// Decode into a `StringWriter`.
```

Comment:
While I'm open to the convenience methods, I think this should clarify that `input` cannot be partial input. That was why I didn't add such methods so far (the proper incremental interface should be used instead).